src/Tools/dotnet-counters/CounterMonitor.cs

```diff
@@ -6,45 +6,32 @@
 using System.CommandLine;
 using System.CommandLine.IO;
 using System.CommandLine.Rendering;
+using System.ComponentModel;
 using System.Diagnostics;
-using System.Diagnostics.Tracing;
-using System.Globalization;
-using System.IO;
 using System.Linq;
-using System.Text;
 using System.Threading;
 using System.Threading.Tasks;
+using Microsoft.Diagnostics.Monitoring;
 using Microsoft.Diagnostics.Monitoring.EventPipe;
 using Microsoft.Diagnostics.NETCore.Client;
 using Microsoft.Diagnostics.Tools.Counters.Exporters;
-using Microsoft.Diagnostics.Tracing;
 using Microsoft.Internal.Common.Utils;
 
 namespace Microsoft.Diagnostics.Tools.Counters
 {
-    public class CounterMonitor
+    internal class CounterMonitor : ICountersLogger
     {
         private const int BufferDelaySecs = 1;
-        private const string SharedSessionId = "SHARED"; // This should be identical to the one used by dotnet-monitor in MetricSourceConfiguration.cs
-        private static HashSet<string> inactiveSharedSessions = new(StringComparer.OrdinalIgnoreCase);
-
-        private string _sessionId;
         private int _processId;
-        private int _interval;
         private CounterSet _counterList;
-        private CancellationToken _ct;
+        private CancellationToken _ct; // We aren't using this, but we do pass "ct" to Start -> should we scrap this, or just reference _ct in Start?
```

Comment:
Let's remove the field if we are not using it anywhere else. Parameters are a lot easier to track than fields